helix-core/src/test/java/org/apache/helix/manager/zk/TestZkHelixAdmin.java

```diff
@@ -560,6 +560,82 @@ public void testLegacyEnableDisablePartition() {
         2);
   }
 
+  @Test
+  public void testResetPartition() throws Exception {
+    String className = TestHelper.getTestClassName();
+    String methodName = TestHelper.getTestMethodName();
+    String clusterName = className + "_" + methodName;
+    String instanceName = "TestInstance";
+    String testResource = "TestResource";
+    String wrongTestInstance = "WrongTestInstance";
+    System.out.println("START " + clusterName + " at " + new Date(System.currentTimeMillis()));
+    HelixAdmin admin = new ZKHelixAdmin(_gZkClient);
+    admin.addCluster(clusterName, true);
+    admin.addInstance(clusterName, new InstanceConfig(instanceName));
+    admin.enableInstance(clusterName, instanceName, true);
+
+    InstanceConfig instanceConfig = admin.getInstanceConfig(clusterName, instanceName);
+
+    // Test the sanity check for resetPartition.
+    // resetPartition is expected to throw an exception when provided with a nonexistent instance.
+    try {
+      admin.resetPartition(clusterName, wrongTestInstance, testResource, Arrays.asList("1", "2"));
```

Comment:
This test case works but not very clean. Since the testResource has not been created, the resetPartition may fail due to other reasons. So the test logic here might be impacted if we change that logic. For example, we check the resource existence first in the future.

To avoid that impact, let's have the resource created and then test reset.
BTW, in that case, you will need to use the verifier to wait until resource assignment has been done and state converged.